src/API.CRUD/Extensions/IMvcCoreBuilderExtensions.cs

```diff
@@ -0,0 +1,58 @@
+using System;
+using API.CRUD.ActionFilters;
+using FluentValidation.AspNetCore;
+using Microsoft.Extensions.DependencyInjection;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Converters;
+
+namespace API.CRUD.Extensions
+{
+    public static class IMvcCoreBuilderExtensions
+    {
+        public static IMvcCoreBuilder AddCustomFluentValidation(this IMvcCoreBuilder mvcCoreBuilder)
+        {
+            if (mvcCoreBuilder == null)
+            {
+                throw new ArgumentNullException(nameof(mvcCoreBuilder));
+            }
+
+            return mvcCoreBuilder.AddFluentValidation(options =>
+                                 {
+                                     options.RegisterValidatorsFromAssemblyContaining<Startup>();
+                                     options.RunDefaultMvcValidationAfterFluentValidationExecutes = false;
+                                 })
+                                 .AddMvcOptions(options => options.Filters.Add(new ValidateModelAttribute()));
+        }
+
+        public static IMvcCoreBuilder AddCustomJsonFormatters(this IMvcCoreBuilder mvcCoreBuilder)
+        {
+            return mvcCoreBuilder.AddJsonFormatters(options =>
+            {
+                options.NullValueHandling = NullValueHandling.Ignore;
+                options.Formatting = Formatting.Indented;
+                options.Converters.Add(new StringEnumConverter());
+            });
+        }
+
+        public static IMvcCoreBuilder AddCustomCors(this IMvcCoreBuilder mvcCoreBuilder,

```

Comment:
We need to move some of these extensions/helper to a common package, so it can be chosen when we need it. CORS for example, isn't needed for our microservices (won't be consumed by browser, only machines). So it's just middleware overhead if it's not needed.

Let's leave this here for now, since it's still early..but make a note to not use the `.AddCustomCors` call to our `Startup` unless it's needed. (only will be needed for the API Gateway)

Reply:
Remove. Add manually if needed. Most microservices won't be called by AJAX.

Reply:
:wrench: